### ISSUE TITLE
Crash on Mach-O universal binary

### ISSUE THREAD

**Issue by llvmx** [revng/revng#327]
It seems to me there is no support for mach-o binaries. Perhaps only for object files. 
The parser always assumes the file is a mach-o object file I think.

Is there any way I can use revng to lift the binary to LLVM IR? 
I mostly care about getting the bitcode, regardless of whether it can be recompiled.

**Comment by aleclearmind** [comment]
Mach-O binaries are supported, despite in a preliminary form.
For instance, debug info are not imported, even though they are in DWARF format, which we support quite extensively.

Please attach a reproducer.

We used to have a working toolchain able to produce Mach-O on Linux, but it bit rot a bit. We need to revive it.
The main reason for this situation is that we can't redistribute it: most of the toolchain can be built from open source software, but not all of it.

**Comment by llvmx** [comment]
What is meant by "preliminary form" in this instance?
I'm trying to lift the binaries on Linux and I got a casting error [To = ObjectFile, From = Binary] for any binary I have tried so far. 
Do I need to install revng on mac and lift the binary there?
Attached binary:
[IMDAppleServices.zip](https://github.com/revng/revng/files/12623549/IMDAppleServices.zip)

**Comment by aleclearmind** [comment]
OK, it was a Mach-O universal binary, which we currently do not support.
I'll soon push a change to provide a better error message.

A quick google search seems to suggest that there's a `lipo` tool that enables you to extract one specific flavor from the universal binary, which should then work. If it doesn't feel free to open a new issue.

We plan to support multiple binaries in a single project. Once that's implemented we can consider directly handling MachO universal files.

**Comment by aleclearmind** [comment]
Fixed in a11435b.

```
$ revng artifact Lift --analyses-list=revng-initial-auto-analysis IMDAppleServices
Unsupported format: MachO universal binary.
```

As said, we'll get back to MachO universal binaries once we support having multiple binaries in a single model (i.e., "project").

Feel free to open a new issue if any of the inner binaries have any problem.
